### PR TITLE
MCOL-4114 Add python3 as dependency

### PR DESCRIPTION
**Pull request by jmrojas2332**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -396,9 +396,9 @@ IF (INSTALL_LAYOUT)
             set(SUSE_VERSION_NUMBER "${CMAKE_MATCH_1}")
         ENDIF ()
         if (${SUSE_VERSION_NUMBER} EQUAL 12)
-            SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "expect" "boost-devel >= 1.54.0" "snappy" "jemalloc" "net-tools MariaDB-server" PARENT_SCOPE)
+            SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "expect" "boost-devel >= 1.54.0" "snappy" "jemalloc" "net-tools" "MariaDB-server" "python3" PARENT_SCOPE)
```
Comment (jmrojas2332) [inline CMakeLists.txt:399]: Not sure if these code paths are ever used, but won't hurt to add dependency here as well

File: cmake/cpackEngineDEB.cmake [hunk 2]
```diff
@@ -56,11 +56,11 @@ if (EXISTS "/etc/debian_version")
...
 endif ()
 if ("${DEBIAN_VERSION_NUMBER}" EQUAL "8")
-    SET(CPACK_DEBIAN_PLATFORM_PACKAGE_DEPENDS "openssl, file, libdbi-perl, libreadline-dev, rsync, net-tools, libboost-all-dev, libsnappy1, libjemalloc1")
+    SET(CPACK_DEBIAN_PLATFORM_PACKAGE_DEPENDS "openssl, file, libdbi-perl, libreadline-dev, rsync, net-tools, libboost-all-dev, libsnappy1, libjemalloc1, MariaDB-server, python3")
 elseif ("${DEBIAN_VERSION_NUMBER}" EQUAL "9")
-    SET(CPACK_DEBIAN_PLATFORM_PACKAGE_DEPENDS "openssl, file, libdbi-perl, libreadline-dev, rsync, net-tools, libboost-all-dev, libsnappy1v5, libreadline5, libjemalloc1")
+    SET(CPACK_DEBIAN_PLATFORM_PACKAGE_DEPENDS "openssl, file, libdbi-perl, libreadline-dev, rsync, net-tools, libboost-all-dev, libsnappy1v5, libreadline5, libjemalloc1, MariaDB-server, python3")
 else()
-    SET(CPACK_DEBIAN_PLATFORM_PACKAGE_DEPENDS "openssl, file, libdbi-perl, libboost-all-dev, libreadline-dev, rsync, libsnappy1v5, net-tools, libjemalloc1")
+    SET(CPACK_DEBIAN_PLATFORM_PACKAGE_DEPENDS "openssl, file, libdbi-perl, libboost-all-dev, libreadline-dev, rsync, libsnappy1v5, net-tools, libjemalloc1, MariaDB-server, python3")
```
Comment (jmrojas2332) [inline cmake/cpackEngineDEB.cmake:63]: require python3 on all deb-based systems

File: cmake/cpackEngineRPM.cmake [hunk 3]
```diff
@@ -77,13 +77,13 @@ IF (EXISTS "/etc/SuSE-release")
...
-    SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "MariaDB-columnstore-shared" "snappy" "net-tools" "MariaDB-server")
+    SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "MariaDB-columnstore-shared" "snappy" "net-tools" "MariaDB-server" "python3")
     # Disable auto require as this will also try to pull Boost via RPM
     SET(CPACK_RPM_PACKAGE_AUTOREQPROV " no")
 elseif (${SUSE_VERSION_NUMBER} EQUAL 12)
-    SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "boost-devel >= 1.54.0" "libsnappy1" "jemalloc" "net-tools" "MariaDB-server")
+    SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "boost-devel >= 1.54.0" "libsnappy1" "jemalloc" "net-tools" "MariaDB-server" "python3")
 else ()
-    SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "boost >= 1.53.0" "snappy" "jemalloc" "net-tools" "MariaDB-server")
+    SETA(CPACK_RPM_columnstore-engine_PACKAGE_REQUIRES "boost >= 1.53.0" "snappy" "jemalloc" "net-tools" "MariaDB-server" "python3")
```
Comment (jmrojas2332) [inline cmake/cpackEngineRPM.cmake:86]: require python3 on all rpm-based systems

File: oam/install_scripts/CMakeLists.txt [hunk 4]
```diff
@@ -22,13 +22,6 @@ configure_file("${CMAKE_CURRENT_SOURCE_DIR}/mcs-ddlproc.service.in" "${CMAKE_CUR
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/mcs-loadbrm.service.in" "${CMAKE_CURRENT_SOURCE_DIR}/mcs-loadbrm.service" @ONLY)
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/mcs-storagemanager.service.in" "${CMAKE_CURRENT_SOURCE_DIR}/mcs-storagemanager.service" @ONLY)
 configure_file("${CMAKE_CURRENT_SOURCE_DIR}/mcs-stop-controllernode.sh.in" "${CMAKE_CURRENT_SOURCE_DIR}/mcs-stop-controllernode.sh" @ONLY)
-
-IF(RPM MATCHES "(rhel|centos)8|(sles|suse)15|fedora")
-    SET(PYTHON_SHEBANG "#!/usr/bin/env python3")
-ELSE()
-    SET(PYTHON_SHEBANG "#!/usr/bin/env python")
-ENDIF()
-
```
Comment (jmrojas2332) [inline oam/install_scripts/CMakeLists.txt:31]: this conditional can go away now due to always using python3

**Comment by mariadb-RomanNavrotskiy** [comment]
please look at
https://ci.columnstore.mariadb.net/mariadb-corporation/mariadb-columnstore-engine/218/2/6

centos7 and all deb platforms have failed smoke test with:

`ERROR 1815 (HY000) at line 1: Internal error: CAL0001: dmlprocessor.cpp PackageHandler::run() package type 0 IDB-2006: 'test.t1' does not exist in Columnstore. `

File: CMakeLists.txt (hunk 1 above)
Comment (drrtuy) [reply CMakeLists.txt:399]: They are used unfortunately. We need to drop them in favor of those in server's storage/columnstore/CMakeLists.txt for RPM and server's debian/control.

File: cmake/cpackEngineDEB.cmake (hunk 2 above)
Comment (drrtuy) [reply cmake/cpackEngineDEB.cmake:63]: This DEB cpack is absolete now. MDB uses its autobake that doesn't use these rules.

File: oam/install_scripts/CMakeLists.txt (hunk 4 above)
Comment (drrtuy) [reply oam/install_scripts/CMakeLists.txt:31]: Don't forget to remove template macroses from mcs-storagemanager.py and make related changes in .gitignore.

**Review by pleblanc1976 — APPROVED** [review]

**Pull request merged**